geomesa-core/src/main/scala/org/locationtech/geomesa/core/iterators/DensityIterator.scala

```diff
@@ -192,28 +138,7 @@ class DensityIterator(other: DensityIterator, env: IteratorEnvironment) extends
     result.put(y, x, cur + 1L)
   }
 
-  override def seek(range: ARange,
-                    columnFamilies: ju.Collection[ByteSequence],
-                    inclusive: Boolean): Unit = {
-    curRange = range
-    source.seek(range, columnFamilies, inclusive)
-    findTop()
-  }
-
-  def hasTop: Boolean = topDensityKey.nonEmpty
-
-  def getTopKey: Key = topDensityKey.orNull
-
-  def getTopValue = topDensityValue.orNull
-
-  def deepCopy(env: IteratorEnvironment): SortedKeyValueIterator[Key, Value] = new DensityIterator(this, env)
-
-  def next(): Unit = if(!source.hasTop) {
-    topDensityKey = None
-    topDensityValue = None
-  } else {
-    findTop()
-  }
+  override def deepCopy(env: IteratorEnvironment): SortedKeyValueIterator[Key, Value] = new DensityIterator(this, env)
```

Comment:
deepCopy should call init on the iterator it's returning. Can't quite tell if that is happening somewhere or not...